test/quantization/fx/test_model_report_fx.py

```diff
@@ -431,3 +436,290 @@ def forward(self, x):
+                self.obs1 = ModelReportObserver()
+                self.fc1 = torch.nn.Linear(5, 5).to(dtype=torch.float)
+                self.relu = torch.nn.ReLU()
+                self.obs2 = ModelReportObserver()
+
+            def forward(self, x):
+                x = self.obs1(x)
+                x = self.fc1(x)
+                x = self.relu(x)
+                x = self.obs2(x)
+                return x
+
+        class LargerIncludeNestModel(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.obs1 = ModelReportObserver()
+                self.nested = TinyNestModule()
+                self.fc1 = SingleLayerLinearModel()
+                self.relu = torch.nn.ReLU()
+
+            def forward(self, x):
+                x = self.obs1(x)
+                x = self.nested(x)
+                x = self.fc1(x)
+                x = self.relu(x)
+                return x
+
+        class ThreeOps(torch.nn.Module):
+            def __init__(self, batch_norm_dim):
+                super(ThreeOps, self).__init__()
+                self.obs1 = ModelReportObserver()
+                self.linear = torch.nn.Linear(7, 3, 2)
+                self.obs2 = ModelReportObserver()
+
+                if batch_norm_dim == 2:
+                    self.bn = torch.nn.BatchNorm2d(2)
+                elif batch_norm_dim == 3:
+                    self.bn = torch.nn.BatchNorm3d(4)
+                else:
+                    raise ValueError("Dim should only be 2 or 3")
+
+                self.relu = torch.nn.ReLU()
+
+            def forward(self, x):
+                x = self.obs1(x)
+                x = self.linear(x)
+                x = self.obs2(x)
+                x = self.bn(x)
+                x = self.relu(x)
+                return x
+
+        class HighDimensionNet(torch.nn.Module):
+            def __init__(self):
+                super(HighDimensionNet, self).__init__()
+                self.obs1 = ModelReportObserver()
+                self.fc1 = torch.nn.Linear(3, 7)
+                self.block1 = ThreeOps(3)
+                self.fc2 = torch.nn.Linear(3, 7)
+                self.block2 = ThreeOps(3)
+                self.fc3 = torch.nn.Linear(3, 7)
+
+            def forward(self, x):
+                x = self.obs1(x)
+                x = self.fc1(x)
+                x = self.block1(x)
+                x = self.fc2(x)
+                y = self.block2(x)
+                y = self.fc3(y)
+                z = x + y
+                z = F.relu(z)
+                return z
+
+        # the purpose of this test is to give the observers a variety of data examples
+        # initialize the model
+        models = [self.NestedModifiedSingleLayerLinear(), LargerIncludeNestModel(), ThreeOps(2), HighDimensionNet()]
+
+        # get random number of epochs and batches
+        num_epochs = torch.randint(1, 10, (1,))  # cap epochs at 10
+        num_batches = torch.randint(1, 15, (1,))  # cap batches at 15
```

Comment:
Not sure if this is a good idea. If this test starts failing for some reason it may be hard to reproduce it